### PR TITLE
Add support for GitHub Actions

### PR DESCRIPTION
**Pull request by greysteil**
GitHub Actions are composable - you can [reference other actions](https://help.github.com/en/articles/configuring-a-workflow#referencing-actions-in-your-workflow) from within a workflow file. To improve the version control story for such references, it would be nice for Dependabot to support updating the version that you reference. For example it would update lines like `uses: actions/setup-node@v1.0.2` to `uses: actions/setup-node@v1.1.0`.

This PR adds support for exactly that. If a `github_actions` update config is enabled, Dependabot will parse your workflow files looking for references to other actions stored on GitHub (i.e., ones references with the format `actions/setup-node@v1.0.1`). If it sees any that are using version-like refs it will check whether those refs can be updated.

A couple of extensions will be wanted in future (**NOT included in this PR**):
- Support for updating versions in workflow files that are imported via `uses: ./some/file`
- Support for updating references to a specific commit (if the default branch includes it and is now ahead of it)
- Support for updating references to a Docker container
- More careful consideration of what to do if a user specifies a `directory` when creating the update config (as GitHub Actions doesn't have monorepo support)

Other than the above, this PR is pretty complete. It's shippable as-is.

### CODE REVIEW

**Comment by greysteil** [comment]
@hfaulds thanks for your help with Actions questions. If you have a moment to scan through this I'd love any feedback. Otherwise I'll get it shipped and iterate it as/when we see any issues.

**Comment by greysteil** [comment]
@rebelagentm I've tagged you as a reviewer so you can see support for a simple language end-to-end

File: github_actions/lib/dependabot/github_actions/file_fetcher.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require "dependabot/file_fetchers"
+require "dependabot/file_fetchers/base"
+
+module Dependabot
+  module GithubActions
+    class FileFetcher < Dependabot::FileFetchers::Base
+      def self.required_files_in?(filenames)
+        filenames.any? { |f| f.match?(%r{\.github/workflows/.+\.ya?ml}) }
```
Comment (jclem) [inline github_actions/lib/dependabot/github_actions/file_fetcher.rb]: FWIW The expression we use `\A\.github/workflows/[^/]+\.ya?ml\z`.

I don't remember why we chose to support only files in the top `.github/workflows` directory, and not, say `.github/workflows/ci/main.yml`, but that's the current situation.

**Comment by xt0rted** [comment]
@greysteil this ran on one of my repos last night and it correctly bumped the version from `v1` to `v1.1.1`, but there wasn't release notes with it. Is that because `v1` would pull `v1.1.1` so it's not treated as an upgrade or is this just not supported right now?

This is the action in question https://github.com/actions/setup-node/releases

**Comment by greysteil** [comment]
It does look like that's caused by the `v1` tag  being moved. Now that you're pinned to a permanent tag  (`v1.1.1` shouldn't move) it won't be a problem for you anymore. I'll have a think about how I can fix it more generally, but it's tricky if repos move the commits their releases point to.

**Comment by xt0rted** [comment]
I forgot you can do that, so yea it makes sense why this wouldn't be showing now. I think having the notes for the version you're being pinned to would be sufficient here. If you were using `v1` then your runs were using the latest version anyhow so the changes prior to pinning probably aren't that relevant.

**Comment by greysteil** [comment]
Yeah, digging into it however I think this is just a case of Dependabot's code being bad and I should feel bad. The `ReleaseFinder` class really phones it in when dealing with dependencies using git tags as their version - I'm fixing that now.

**Comment by greysteil** [comment]
Fixed by https://github.com/dependabot/dependabot-core/commit/7e68d4f4290a5cb29c8203e4c4ba54a5683b0526. Thanks for the ping!

**Comment by xt0rted** [comment]
Thanks for the awesome tool and getting this added in so quickly!

**Comment by WtfJoke** [comment]
Cool that you already enabled support for github actions. 👍 

Two questions: 
1. How do I enable it in config.yml? 
2. When is the github action support enabled again? Or why was it disabled in first place?

Github Actions Support currently disabled

**Pull request merged**